packages/graphql-live-query-patch/src/createLiveQueryPatchInflator.ts

```diff
@@ -0,0 +1,57 @@
+import { applyPatch } from "fast-json-patch";
+import { ExecutionResult } from "graphql";
+import { LiveExecutionPatch } from "./LiveExecutionPatch";
+
+export async function* createLiveQueryPatchInflator(
+  asyncIterator: AsyncIterableIterator<LiveExecutionPatch & ExecutionResult>
+) {
+  let previousValue: ExecutionResult | null = null;
+  let previousRevision = 0;
+
+  for await (const result of asyncIterator) {
+    if (result.isLivePatch) {
+      if (!result.revision) {
+        throw new Error("Missing revision for patch result.");
+      }
+      if (result.data) {
+        previousValue = { data: result.data } as ExecutionResult;
+        if (result.extensions) {
+          previousValue.extensions = result.extensions;
+        }
+        if (result.errors) {
+          previousValue.errors = result.errors;
+        }
+        previousRevision = result.revision;
+
+        yield previousValue as ExecutionResult;
+      } else if (result.patch) {
+        if (!previousValue) {
+          throw new Error("No previousValue available.");
+        }
+        if (previousRevision + 1 !== result.revision) {
+          throw new Error("Wrong revision received.");
```

Comment:
I was anticipating that revision numbers were here to handle transport layers that do not guarantee message delivery order but this seems to negate that assumption as it will throw an error if an out of order message is received.

Are revision numbers redundant on in-order delivery transports or are they facilitating something I'm not seeing yet?

PS. This looks great so far. Kudos on doing the thing!

Reply:
I added them to ensure data consistency, e.g. when a single patch somehow got lost then it would not make sense to apply any further patches as they could further disrupt the state on the client. However I have not encountered a situation where a message was lost

Reply:
It seems like WebSocket enforces sequential order, however with HTTP polling that wouldn't be guaranteed.

Reply:
ah, HTTP polling was the thing I was missing. Yeah, good point. :+1:

Reply:
As a follow up to that when it comes time to write a spec for this we should leave `result.revision` as an optional field so that in a future optimization it could be omitted on transports that already enforce sequential order.

Reply:
So the idea would be to create a simple lookup map where results are stored that are not received in chronological order; then we need some mechanism for actually failing in case sth is not arriving. Something like a configurable threshold of few seconds seems fine with me. Then when a single result arrives we can check whether all/some results can be applied/flushed in bulk.

Reply:
gonna fix that later, wanna pus a first version to play with out there rn!